fatrace.c

```diff
@@ -251,6 +253,12 @@ print_event (const struct fanotify_event_metadata *data,
                 len--;
             procname[len] = '\0';
             procname_pid = data->pid;
+
+            /* get user and group */
+            if (option_user == 1) {
+                if (fstat (proc_fd, &ug) < 0)
+                    err (EXIT_FAILURE, "uid/gid stat");
+            }
             got_procname = true;
```

Comment:
Please keep this after setting `procname_pid`, so that related code stays together.

Reply:
Reworked, see below.